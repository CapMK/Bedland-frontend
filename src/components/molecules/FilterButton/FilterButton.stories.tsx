import type { Meta, StoryObj } from '@storybook/react';
import ThemeProvider from 'theme/ThemeContext';
import GlobalStyle from 'theme/globalStyles';
import FilterButton from './FilterButton';

const meta = {
  title: 'Molecules/FilterButton',
  component: FilterButton,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
  parameters: {
    backgrounds: { default: 'dark' },
  },
} satisfies Meta<typeof FilterButton>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
