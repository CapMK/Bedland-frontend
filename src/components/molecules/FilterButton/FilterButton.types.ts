import { ButtonVariantProps, HoverVariant } from 'components/atoms/Button/Button.types';

type PossibleButtons = {
  primaryBorders: {
    buttonType: ButtonVariantProps;
    text: string;
    hover: HoverVariant;
  };
  primary: {
    buttonType: ButtonVariantProps;
    text: string;
    hover: HoverVariant;
  };
};

export default PossibleButtons;
