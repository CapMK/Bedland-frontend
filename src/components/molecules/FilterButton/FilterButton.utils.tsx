import PossibleButtons from './FilterButton.types';

export const possibleButtons: PossibleButtons = {
  primaryBorders: {
    buttonType: 'primaryBorders',
    text: 'Filters selected',
    hover: 'default',
  },
  primary: {
    buttonType: 'primary',
    text: 'Select filters',
    hover: 'text',
  },
};

export const whichButton = (ifActive: boolean) =>
  ifActive ? possibleButtons.primaryBorders : possibleButtons.primary;
