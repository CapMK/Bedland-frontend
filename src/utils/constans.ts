export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
