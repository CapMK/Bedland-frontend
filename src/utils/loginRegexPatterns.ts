import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
