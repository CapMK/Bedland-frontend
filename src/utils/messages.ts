const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
