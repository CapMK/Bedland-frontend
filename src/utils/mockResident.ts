const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
