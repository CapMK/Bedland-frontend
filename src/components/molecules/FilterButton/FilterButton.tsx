import Button from 'components/atoms/Button/Button';
import { useState } from 'react';
import { whichButton } from './FilterButton.utils';

const FilterButton = () => {
  const [ifActive, setIfActive] = useState<boolean>(false);
  const { buttonType, text } = whichButton(ifActive);
  const handleFilterButton = () => setIfActive(!ifActive);

  return (
    <Button wrapper>
      <Button
        variant={buttonType}
        color='white'
        hover='text'
        icon='filterButton'
        onClick={handleFilterButton}
      >
        {text}
      </Button>
    </Button>
  );
};

export default FilterButton;
