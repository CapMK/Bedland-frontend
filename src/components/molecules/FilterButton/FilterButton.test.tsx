import { cleanup } from '@testing-library/react';
import { renderWithProviders } from 'utils/test.utils';
import FilterButton from './FilterButton';
import { whichButton, possibleButtons } from './FilterButton.utils';

describe('Filter Button', () => {
  afterEach(cleanup);
  const filterButton = () => renderWithProviders(<FilterButton />, {});

  it('filterButton renders correctly', () => {
    expect(filterButton()).toMatchSnapshot();
  });

  it('whichButton function should return primaryButton', () => {
    expect(whichButton(false)).toEqual(possibleButtons.primary);
  });

  it('whichButton function should return primaryBordersButton', () => {
    expect(whichButton(true)).toEqual(possibleButtons.primaryBorders);
  });
});
