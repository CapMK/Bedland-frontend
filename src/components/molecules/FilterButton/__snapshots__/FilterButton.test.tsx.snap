// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filter Button > filterButton renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-radius: 0.5rem;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c0:hover {
  background-color: #fff;
  border: 1px solid #fff;
  color: #6573FF;
}

<body>
    <div>
      <div
        class=""
      >
        <button
          class="c0"
        >
          <svg
            data-testid="filterButton-icon"
            fill="none"
            height="20"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-filterButton"
            />
            <path
              d="M13.0002 1.3335H3.00023C2.07939 1.3335 1.31189 2.09183 1.53939 2.9835C1.78159 3.92841 2.22773 4.80883 2.84649 5.56292C3.46526 6.31701 4.24162 6.92647 5.12106 7.3485C5.80689 7.67683 6.33356 8.32183 6.33356 9.08183V14.1518C6.33364 14.2938 6.36999 14.4334 6.43918 14.5574C6.50837 14.6814 6.60809 14.7857 6.72889 14.8603C6.84969 14.935 6.98755 14.9775 7.1294 14.9839C7.27125 14.9903 7.41238 14.9603 7.53939 14.8968L9.20606 14.0635C9.34446 13.9943 9.46088 13.888 9.54227 13.7564C9.62366 13.6249 9.66681 13.4732 9.66689 13.3185V9.08183C9.66689 8.32183 10.1936 7.67683 10.8786 7.3485C11.7583 6.92671 12.5351 6.31735 13.1541 5.56324C13.7732 4.80914 14.2196 3.92859 14.4619 2.9835C14.6877 2.09183 13.9194 1.3335 13.0002 1.3335Z"
              stroke="#1B1A1A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.67"
            />
          </svg>
          Select filters
        </button>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-radius: 0.5rem;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c0:hover {
  background-color: #fff;
  border: 1px solid #fff;
  color: #6573FF;
}

<div>
    <div
      class=""
    >
      <button
        class="c0"
      >
        <svg
          data-testid="filterButton-icon"
          fill="none"
          height="20"
          viewBox="0 0 16 16"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-filterButton"
          />
          <path
            d="M13.0002 1.3335H3.00023C2.07939 1.3335 1.31189 2.09183 1.53939 2.9835C1.78159 3.92841 2.22773 4.80883 2.84649 5.56292C3.46526 6.31701 4.24162 6.92647 5.12106 7.3485C5.80689 7.67683 6.33356 8.32183 6.33356 9.08183V14.1518C6.33364 14.2938 6.36999 14.4334 6.43918 14.5574C6.50837 14.6814 6.60809 14.7857 6.72889 14.8603C6.84969 14.935 6.98755 14.9775 7.1294 14.9839C7.27125 14.9903 7.41238 14.9603 7.53939 14.8968L9.20606 14.0635C9.34446 13.9943 9.46088 13.888 9.54227 13.7564C9.62366 13.6249 9.66681 13.4732 9.66689 13.3185V9.08183C9.66689 8.32183 10.1936 7.67683 10.8786 7.3485C11.7583 6.92671 12.5351 6.31735 13.1541 5.56324C13.7732 4.80914 14.2196 3.92859 14.4619 2.9835C14.6877 2.09183 13.9194 1.3335 13.0002 1.3335Z"
            stroke="#1B1A1A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.67"
          />
        </svg>
        Select filters
      </button>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
